conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/DialogueFeignClient.java

```diff
@@ -0,0 +1,346 @@
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.client.jaxrs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Multimaps;
+import com.google.common.io.CharStreams;
+import com.google.common.net.HttpHeaders;
+import com.google.common.primitives.Ints;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import com.palantir.conjure.java.api.errors.RemoteException;
+import com.palantir.conjure.java.api.errors.SerializableError;
+import com.palantir.conjure.java.api.errors.UnknownRemoteException;
+import com.palantir.conjure.java.serialization.ObjectMappers;
+import com.palantir.dialogue.Channel;
+import com.palantir.dialogue.Clients;
+import com.palantir.dialogue.ConjureRuntime;
+import com.palantir.dialogue.Deserializer;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.HttpMethod;
+import com.palantir.dialogue.RequestBody;
+import com.palantir.dialogue.Response;
+import com.palantir.dialogue.UrlBuilder;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.UnsafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+import feign.Request;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * {@link DialogueFeignClient} is an adapter from {@link feign.Client} to {@link Channel Dialogue Channel}
+ * taking advantage of the superior observability and stability provided by Dialogue.
+ */
+final class DialogueFeignClient implements feign.Client {
+
+    private static final String PATH_TEMPLATE = "hr-path-template";
+    private static final Splitter pathSplitter = Splitter.on('/').omitEmptyStrings();
+    private static final Splitter.MapSplitter querySplitter =
+            Splitter.on('&').omitEmptyStrings().withKeyValueSeparator('=');
+
+    private final Clients clients;
+    private final Channel channel;
+    private final String baseUrl;
+    private final String serviceName;
+    private final String version;
+
+    DialogueFeignClient(Class<?> jaxrsInterface, Channel channel, ConjureRuntime runtime, String baseUrl) {
+        this.channel = Preconditions.checkNotNull(channel, "Channel is required");
+        this.serviceName = jaxrsInterface.getSimpleName();
+        this.version = Optional.ofNullable(jaxrsInterface.getPackage().getImplementationVersion())
+                .orElse("0.0.0");
+        this.baseUrl = baseUrl;
+        this.clients = runtime.clients();
```

Comment:
could we just store the whole ConjureRuntime as a field? on the first read through the `Clients clients` thing made me worry this was doing some kind of internal caching!